kitsune/users/utils.py

```diff
@@ -159,3 +165,25 @@ def user_is_contributor(user):
         user.is_authenticated
         and user.groups.filter(name__in=ContributionAreas.get_groups()).exists()
     )
+
+
+def delete_user_pipeline(user: User) -> None:
+    """
+    Deletes a user and all associated data.
+    Returns True if the user was deleted successfully, False otherwise.
+    """
+
+    publisher = UserDeletionPublisher(user=user)
+
+    publisher.register_listener(ThreadListener())
+    publisher.register_listener(PostListener())
+    publisher.register_listener(KBThreadListener())
+    publisher.register_listener(KBPostListener())
+    publisher.register_listener(AAQListener())
+    publisher.register_listener(DocumentListener())
+
+    publisher.publish()
+
+    # if all listeners have completed processing, delete the user
+    if publisher.completed_processing():
```

Comment:
Per my comment above, I don't see any need for this call to `completed_processing()`. As far as I can tell, none of the work of any of the listeners called within `publish()` is ever done asynchronously.